File: integrators/fastapi/qelos_integrator_fastapi/proxy_target.py
from __future__ import annotations

import os

from .config import QelosConfig


def resolve_qelos_proxy_target(config: QelosConfig) -> str | None:
    """Resolve the origin to which ``/api/**`` requests are proxied.

    Priority (highest first):

    1. ``QELOS_PROXY_TARGET`` env var
    2. ``QELOS_IP`` env var
    3. ``QELOS_API_IP`` env var
    4. ``config.app_url``

    Empty / whitespace-only values are skipped. Returns ``None`` when nothing
    is configured so callers can return 503 or treat the request as anonymous.
    """
    env = os.environ
    from_env = (
        (env.get("QELOS_PROXY_TARGET") or "").strip()
        or (env.get("QELOS_IP") or "").strip()
        or (env.get("QELOS_API_IP") or "").strip()
    )
    if from_env:
        return from_env

    from_app_url = (config.app_url or "").strip()
    return from_app_url or None

File: integrators/next/src/proxy-target.ts
import type { QelosNextConfig } from './types';

/**
 * Resolve the origin to which `/api/**` requests are proxied. The managed
 * Qelos app URL (`config.appUrl`) is the proxy target — env vars are only
 * dev-time overrides when the configured `appUrl` is not reachable from
 * the local host.
 *
 * Priority (highest first):
 *   1. `process.env.NEXT_QELOS_PROXY_TARGET`
 *   2. `process.env.QELOS_IP`
 *   3. `process.env.QELOS_API_IP`
 *   4. `config.appUrl`
 *
 * Empty / whitespace-only values are skipped, and `undefined` is returned when
 * nothing is configured so the caller can surface a clear error.
 */
export function resolveQelosProxyTarget(
  config: QelosNextConfig,
): string | undefined {
  const env = process.env;
  const fromEnv =
    env.NEXT_QELOS_PROXY_TARGET?.trim() ||
    env.QELOS_IP?.trim() ||
    env.QELOS_API_IP?.trim();
  if (fromEnv) return fromEnv;

  const fromAppUrl = config.appUrl?.trim();
  return fromAppUrl || undefined;
}
